configure.ac

```diff
@@ -571,10 +571,28 @@ AS_IF([test "$with_systemd" != "no"], [
 
 AM_CONDITIONAL(WITH_SYSTEMD, [test "$with_systemd" = "yes"])
 
+# --------------------------------------------------------------------
+# bash completion
+
+AC_ARG_WITH(
+	[bashcompdir],
+	[AS_HELP_STRING([--with-completiondir=PATH],
+			[Directory of Bash completion @<:@detect@:>@])])
+
+AS_IF([test "${bashcompdir}" != "no"], [
+	PKG_CHECK_MODULES([BASH_COMPLETION], [bash-completion >= 2.0], [],
+		[bashcompdir=no])
+
+	PKG_CHECK_VAR(bashcompdir, [bash-completion], [completionsdir], [],
+		[bashcompdir="${sysconfdir}/bash_completion.d"])
```

Comment:
`bashcompdir='$(datadir)/bash-completion'`

Reply:
Can you clarify here? It is either `/usr/share/bash-completion/completions/` (as from the pkgconfig `pkg-config --variable=completionsdir bash-completion`) or the legacy location under etc (`/etc/bash_completion.d`) -- safe fallback if the pkgconfig does not work or something.

Reply:
FWIW the intent in our (bash-completion) autotools recipe is exactly what @Jakuje described. In case you want to support bash-completion 1.x users, `/etc/bash_completion.d` is the dir to use, and as it doesn't have pkgconfig support, that's where the fallback will take the file. It'll work with more modern bash-completion as well, in case pkgconfig or bash-completion's `.pc` file are not available.

Reply:
While I understand that, I still don't think we should support the legacy version, as it adds a technical debt. I've checked some other packages and they assume either `.pc` is present or `$(datadir)/bash-completion` is used:
https://github.com/latchset/clevis/blob/master/src/bash/meson.build
https://github.com/flatpak/flatpak/blob/master/Makefile.am#L138

As for the check I would suggest simply doing:
```sh
PKG_CHECK_VAR([bashcompdir], [bash-completion], [with_bash_completion=yes])
AM_CONDITIONAL(WITH_BASH_COMPLETION, [test "$with_bash_completion" = "yes"])
```
and we probably don't want the configure option.

Reply:
I doubt that we would be installing these with old bash completion. So far, I left it as it was, but I can change it to the datadir as a fallback if you will insist.

Reply:
I do not insist on the fallback now, but I rather suggest removing the fallback at all.

Reply:
Ok, makes sense. I tried to update it, so hopefully it will work. As well as the meson target (lets see CI).